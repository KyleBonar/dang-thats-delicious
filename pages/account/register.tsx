import * as React from "react";
import RegisterUser from "../../src/components/RegisterUser/RegisterUser";
import { Article } from "../../src/components/Article/Article";
import HeaderSimple from "../../src/components/HeaderSimple/HeaderSimple";

export interface RegisterProps {}

const Register: React.SFC<RegisterProps> = () => {
  return (
    <>
      <HeaderSimple />
      <Article size="sm">
        <RegisterUser />
      </Article>
    </>
  );
};

export default Register;
