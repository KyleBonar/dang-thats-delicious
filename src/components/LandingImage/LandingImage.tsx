import * as React from "react";
import { useSelector } from "react-redux";
import { useRouter } from "next/router";

import { AppState } from "../../redux/configureStore";
import {
  HeroContainer,
  HeroImage,
  HeroBody,
  HeroTitle,
  StyledDiv,
  StyledDropDown,
  StyledInput,
  StyledButton
} from "./LandingImageStyle";

export interface LandingImageProps {
  className?: string;
}

const LandingImage: React.FunctionComponent<LandingImageProps> = props => {
  const [search, setSearch] = React.useState("");
  const [selectedValue, setSelectedValue] = React.useState("all");
  const types = useSelector((store: AppState) => {
    return store.sauces.types;
  });

  // assign router
  const router = useRouter();

  return (
    <HeroContainer className={props.className}>
      <HeroImage />
      <HeroBody>
        <HeroTitle>Find your perfect sauce</HeroTitle>
        <form onSubmit={onSubmit}>
          <StyledDiv>
            <StyledDropDown
              options={types}
              selectedValue={selectedValue}
              onSelect={e => setSelectedValue(e.target.value)}
            />
            <StyledInput
              type="text"
              id="Hero__Search"
              name="Hero__Search"
              onChange={e => setSearch(e.target.value)}
              placeholder="Search..."
            />

            <StyledButton type="submit">Search</StyledButton>
          </StyledDiv>
        </form>
      </HeroBody>
    </HeroContainer>
  );

  async function onSubmit(event: React.FormEvent): Promise<null> {
    event.preventDefault();

    let query = "/sauces?";
    if (search.length > 0) {
      query += `srch=${search}`;
    }

    if (selectedValue.toLowerCase() !== "all") {
      query += `&type=${selectedValue}`;
    }

    // take person to sauces page w/ prefilled search query
    router.push(query);
    return null;
  }
};

export default LandingImage;
