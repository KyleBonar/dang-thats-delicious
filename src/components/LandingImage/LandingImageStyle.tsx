import styled from "../../theme/styled-components";
import Select from "../Select/Select";
import { TextInput } from "../TextInput/TextInput";
import { Button } from "../Button/Button";

// assign image path
const SaucesImage = "/static/LandingPageImage.jpg";

export const HeroContainer = styled.div`
  background: #000;
  position: relative;
  margin-bottom: 1.5rem;

  @media (min-width: ${props => props.theme.smToMd}) {
    margin-bottom: 3.5rem;
  }
`;
HeroContainer.displayName = "div";

export const HeroImage = styled.div`
  background-image: url('${SaucesImage}');
  background-size: cover;
  background-position-y: 45%;
  opacity: 0.45;
  height: 400px;
`;
HeroImage.displayName = "div";

export const HeroBody = styled.div`
  position: absolute;
  z-index: 1;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  padding: 0 2em;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: center;
`;
HeroBody.displayName = "div";

export const HeroTitle = styled.h1`
  color: ${x => x.theme.landingHeroTextColor};
  font-family: FuturaMedium;
`;
HeroTitle.displayName = "h1";

export const StyledDiv = styled.div`
  display: flex;
  flex-direction: row;
  align-items: stretch;
`;
StyledDiv.displayName = "div";

export const StyledSelect = styled(Select)`
  min-height: 30px;
  border: 0px;
  box-sizing: border-box;

  > div {
    width: auto;

    &:after {
      top: 3px;
    }
  }

  select {
    width: auto;
    border: 0;
  }
`;
StyledSelect.displayName = "select";

export const StyledInput = styled(TextInput)`
  flex-direction: row;
  align-items: stretch;

  input {
    border: 0px;
    margin-top: 0px;
    margin-bottom: 0px;
    width: 100%;
    min-width: auto;
    max-width: 100%;
    padding: 0.66em;
    box-sizing: border-box;
    border: 0px;
    font-size: 1em;

    @media (min-width: ${props => props.theme.smToMd}) {
      min-width: 20em;
    }
  }
`;
StyledInput.displayName = "input";

export const StyledButton = styled(Button)`
  display: flex;
`;
StyledButton.displayName = "button";
