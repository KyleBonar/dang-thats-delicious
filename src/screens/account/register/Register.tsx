import * as React from "react";
import { connect } from "react-redux";

import { register } from "../../../redux/users/actions";
import { IRegisterUser } from "../../../redux/users/types";
import LogoSFYT from "../../../images/icons/LogoSFYT";
import PageTitle from "../../../components/PageTitle/PageTitle";
import { TextInput } from "../../../components/TextInput/TextInput";
import { Link } from "../../../components/Link/Link";
import { AppState } from "../../../redux/configureStore";
import {
  StyledDiv,
  StyledLogoContainer,
  StyledArticle,
  StyledFormContainer,
  StyledText,
  StyledButton
} from "./RegisterStyle";
import {
  FlashMessage,
  FlashMessageProps
} from "../../../components/FlashMessage/FlashMessage";
import Auth from "../../../utils/Auth/Auth";

export interface RegisterProps {
  register: any;
  history: { push: (location: string) => null };
}

export interface RegisterState {
  email: string;
  confirmEmail: string;
  password: string;
  confirmPassword: string;
  displayName: string;
  flashMessage: FlashMessageProps;
}

class Register extends React.Component<RegisterProps, RegisterState> {
  constructor(props: RegisterProps) {
    super(props);

    // Init state
    this.state = {
      email: "",
      confirmEmail: "",
      password: "",
      confirmPassword: "",
      displayName: "",
      flashMessage: {
        isVisible: false
      }
    };
  }

  public render() {
    return (
      <StyledDiv>
        <StyledLogoContainer>
          <Link to="/">
            <LogoSFYT />
          </Link>
        </StyledLogoContainer>
        <hr />
        <StyledArticle>
          <PageTitle>Register</PageTitle>
          <StyledFormContainer>
            {this.state.flashMessage.isVisible && (
              <FlashMessage type={this.state.flashMessage.type} isVisible>
                {this.state.flashMessage.text}
              </FlashMessage>
            )}
            <form onSubmit={this.onSubmit} style={{ width: "100%" }}>
              <TextInput
                type="email"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Email"}
                name={"email"}
                value={this.state.email}
                required={true}
              />
              <TextInput
                type="email"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Confirm Email"}
                name={"confirmEmail"}
                value={this.state.confirmEmail}
                required={true}
                requirementText={"Must match above."}
              />
              <TextInput
                type="password"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Password"}
                name={"password"}
                value={this.state.password}
                required={true}
                requirementText={"Must be at least 9 characters long."}
              />
              <TextInput
                type="password"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Confirm Password"}
                name={"confirmPassword"}
                value={this.state.confirmPassword}
                required={true}
                requirementText={"Must match above."}
              />
              <TextInput
                type="text"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Display Name"}
                name={"displayName"}
                value={this.state.displayName}
                required={true}
                requirementText={"Must be at least 6 characters long."}
              />
              <StyledText>
                By clicking 'Register', you agree to Sauce For Your Thoughts{" "}
                <Link to="#">Terms and Conditions</Link>
              </StyledText>
              <StyledButton type="submit">Register</StyledButton>
            </form>
          </StyledFormContainer>
        </StyledArticle>
      </StyledDiv>
    );
  }

  private onTextChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    if (!event || !event.target) {
      return;
    }
    // Grab the name and value
    const { name, value }: { name: string; value: string } = event.target;

    // Update local state
    this.setState({
      ...this.state,
      [name]: value
    });
  };

  private onSubmit = async (event: React.FormEvent): Promise<any> => {
    event.preventDefault();

    if (this.state.email !== this.state.confirmEmail) {
      this.setState({
        flashMessage: {
          isVisible: true,
          text: "Your emails do not match. Please fix this before continuing.",
          type: "alert"
        }
      });

      window.scrollTo(0, 0); // Move screen to top

      return;
    }

    if (this.state.password !== this.state.confirmPassword) {
      this.setState({
        flashMessage: {
          isVisible: true,
          text:
            "Your passwords do not match. Please fix this before continuing.",
          type: "alert"
        }
      });

      window.scrollTo(0, 0); // Move screen to top

      return;
    }

    if (this.state.password.length < 8) {
      this.setState({
        flashMessage: {
          isVisible: true,
          text:
            "Your password is too weak! Please make your password over 8 characters long.",
          type: "alert"
        }
      });

      window.scrollTo(0, 0); // Move screen to top

      return;
    }

    if (this.state.displayName.length < 6) {
      this.setState({
        flashMessage: {
          isVisible: true,
          text:
            "Your display name is not long enough. It must be at least 6 characters long.",
          type: "alert"
        }
      });

      window.scrollTo(0, 0); // Move screen to top

      return;
    }

    // Grab values from state
    const {
      email,
      confirmEmail,
      password,
      confirmPassword,
      displayName
    } = this.state;

    // Create credentials obj
    const credentials: IRegisterUser = {
      user: { email, confirmEmail, password, confirmPassword, displayName }
    };
    try {
      // dispatch action which calls API to register user
      await this.props.register({ credentials });

      // Redirect user to sauces page -- Maybe take them to user home page instead?
      this.props.history.push("/sauces");
    } catch (err) {
      // Create warning flash
      this.setState({
        flashMessage: {
          isVisible: true,
          text: err.response.data.msg,
          type: "warning"
        }
      });

      window.scrollTo(0, 0); // Move screen to top
    }
  };
}

const mapState2Props = (state: AppState) => {
  return {};
};

const mapDispatch2Props = {
  register
};

export default connect(
  mapState2Props,
  mapDispatch2Props
)(Register);
