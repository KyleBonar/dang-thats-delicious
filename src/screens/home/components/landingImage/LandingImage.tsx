import * as React from "react";

import styled from "../../../../theme/styled-components";
import DropDown from "./components/DropDown/DropDown";
import Input from "./components/Input/Input";
import Button from "../../../../components/Button/Button";

const HeroContainer = styled.header`
  background: #000;
  position: relative;
`;

const HeroImage = styled.div`
  background-image: url("../../../../images/photos/Sauces.jpg");
  background-size: cover;
  background-position-y: 45%;
  opacity: 0.45;
  height: 400px;
`;

const HeroBody = styled.div`
  position: absolute;
  z-index: 1;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  padding: 0 2em;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: center;
`;

const HeroTitle = styled.h1`
  color: ${x => x.theme.landingHeroTextColor};
`;

const Div = styled.div`
  display: flex;
  flex-direction: row;
  align-items: stretch;
`;

export interface LandingImageProps {}

export interface LandingImageState {
  search: {
    value: string;
  };
  filter: {
    all: string[];
    selectedValue: string;
  };
}

class LandingImage extends React.Component<
  LandingImageProps,
  LandingImageState
> {
  constructor(props: LandingImageProps) {
    super(props);

    this.state = {
      search: { value: "" },
      filter: { all: ["All", "Hot Sauce", "Meat Sauce"], selectedValue: "all" }
    };
  }

  public render(): JSX.Element {
    return (
      <HeroContainer>
        <HeroImage />
        <HeroBody>
          <HeroTitle>Find your perfect sauce</HeroTitle>
          <Div>
            <DropDown
              options={this.state.filter.all}
              selectedValue={this.state.filter.selectedValue}
              onSelect={this.onSelect}
            />
            <Input
              id="Hero__Search"
              name="Hero__Search"
              onChange={this.onTextChange}
              value={this.state.search.value}
            />
            <Button
              type="solid"
              isLink={true}
              text="Search Me"
              onClick={this.onClick}
            />
          </Div>
        </HeroBody>
      </HeroContainer>
    );
  }

  private onSelect = (event: React.ChangeEvent<HTMLSelectElement>): void => {
    if (!event || !event.target || !event.target.value) {
      return;
    }

    // Grab the value
    const selectedValue: string = event.target.value;

    // Update local state
    this.setState({
      ...this.state,
      filter: {
        ...this.state.filter,
        selectedValue
      }
    });
  };

  private onTextChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    if (!event || !event.target || !event.target.value) {
      return;
    }

    // Grab the value
    const value: string = event.target.value;

    // Update local state
    this.setState({
      ...this.state,
      search: {
        value
      }
    });
  };

  private onClick = (event: any): void => {
    console.log(event);
  };
}

export default LandingImage;
