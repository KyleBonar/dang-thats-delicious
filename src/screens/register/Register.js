import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import RegisterForm from "./components/registerForm/RegisterForm";

import { flashError, flashClose } from "../../redux/actions/flash";
import { register } from "../../redux/actions/auth";

class Register extends Component {
  static propType = {
    history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
    register: PropTypes.func.isRequired,
    flashError: PropTypes.func.isRequired,
    flashClose: PropTypes.func.isRequired
  };
  render() {
    return (
      <div className="inner">
        <RegisterForm onSubmit={this.handleSubmit} />
      </div>
    );
  }

  handleSubmit = e => {
    const data = {
      user: {
        email: e.email,
        name: e.name,
        password: e.password,
        confirmPassword: e.confirmPassword
      }
    };
    this.props.flashClose();
    this.props
      .register(data)
      .then(() => this.props.history.push("/sauces"))
      .catch(err => {
        console.log(err);
        console.log(err.response);
        this.props.flashError({ text: err.response.data.msg });
      });
  };
}

const mapDispatchToProps = { register, flashError, flashClose };

export default connect(null, mapDispatchToProps)(Register);
