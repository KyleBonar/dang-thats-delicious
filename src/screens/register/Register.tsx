import * as React from "react";
import { connect } from "react-redux";
import LogoSFYT from "../../images/icons/LogoSFYT";
import styled from "styled-components";
import Article from "../../components/Article/Article";
import PageTitle from "../../components/PageTitle/PageTitle";
import TextInput from "../../components/TextInput/TextInput";

const StyledLogoContainer = styled.div`
  max-width: 150px;
  margin: 0 auto;
  padding: 1em;
`;

const StyledArticle = styled(Article)`
  max-width: 600px;
`;
const StyledFormContainer = styled.div`
  background-color: ${props => props.theme.formContainerBackgroundColor};
  padding: 1.5rem;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

export interface RegisterProps {}

export interface RegisterState {}

class Register extends React.Component<RegisterProps, any> {
  public render() {
    return (
      <div>
        <StyledLogoContainer>
          <LogoSFYT />
        </StyledLogoContainer>
        <hr />
        <StyledArticle>
          <PageTitle>Register</PageTitle>
          <StyledFormContainer>
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Email"}
              required={true}
            />
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Confirm Email"}
              required={true}
            />
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Password"}
              required={true}
            />
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Confirm Password"}
              required={true}
            />
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Display Name"}
              required={true}
            />
          </StyledFormContainer>
        </StyledArticle>
      </div>
    );
  }
}

const mapState2Props = state => {
  return {};
};

export default connect(mapState2Props)(Register);
