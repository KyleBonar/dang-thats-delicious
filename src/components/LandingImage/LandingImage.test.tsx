import "jsdom-global/register";
import * as React from "react";
import * as enzyme from "enzyme";
import { Provider } from "react-redux";

import LandingImage, { LandingImageProps } from "./LandingImage";
import {
  ITERATION_SIZE,
  fakeStore,
  casual
} from "../../utils/testUtils/testUtils";

const mockStores = new Array(ITERATION_SIZE).fill(null).map(fakeStore);

describe("<LandingImage />", () => {
  // const _defaultTitleText = "Find your perfect sauce";

  it("renders", () => {
    new Array(ITERATION_SIZE).fill(null).forEach(() => {
      const wrapper = enzyme.shallow(<LandingImage />);

      expect(wrapper).toBeTruthy();
    });
  });

  it("matches snapshot", () => {
    new Array(ITERATION_SIZE).fill(null).forEach(() => {
      const wrapper = enzyme.shallow(<LandingImage />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  it("appends classname to parent", () => {
    new Array(ITERATION_SIZE).fill(null).forEach(() => {
      const className = casual.string;
      const wrapper = enzyme.shallow(<LandingImage className={className} />);

      expect(wrapper.first().prop("className")).toEqual(className);
    });
  });

  // it("renders a form", () => {
  //   mockStores.forEach(mockStore => {
  //     const wrapper = enzyme.shallow(<LandingImage />);

  //     expect(wrapper.find("form").exists()).toBeTruthy();
  //   });
  // });

  // it("renders a default title text", () => {
  //   mockStores.forEach(mockStore => {
  //     const wrapper = enzyme.shallow(<LandingImage />);

  //     expect(wrapper.find("h1").first().text()).toEqual(_defaultTitleText);
  //   });
  // });
});
