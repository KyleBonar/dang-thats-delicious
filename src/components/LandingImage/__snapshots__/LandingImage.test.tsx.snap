// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingImage /> matches snapshot 1`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 2`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 3`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 4`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 5`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 6`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 7`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 8`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 9`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 10`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 11`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 12`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 13`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 14`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 15`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 16`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 17`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 18`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 19`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 20`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 21`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 22`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 23`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 24`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 25`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 26`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 27`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 28`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 29`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 30`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 31`] = `
<div>
  <div />
  <Body />
</div>
`;

exports[`<LandingImage /> matches snapshot 32`] = `
<div>
  <div />
  <Body />
</div>
`;
